sdk/cosmos/azure-spring-data-cosmos/src/main/java/com/azure/spring/data/cosmos/core/generator/AbstractQueryGenerator.java

```diff
@@ -268,9 +268,26 @@ protected SqlQuerySpec generateCosmosQuery(@NonNull CosmosQuery query,
                                                          toCosmosDbValue(p.getSecond())))
                                                      .collect(Collectors.toList());
 
-        if (query.getLimit() > 0) {
+
+        long offset = 0;
+        long pageSize = 0;
+        try {
+                offset = query.getPageable().getOffset();
+                pageSize = query.getPageable().getPageSize();
+        } catch (UnsupportedOperationException ex) {
+            // Ignore
+        }
+
+        if (offset > 0 && pageSize > 0 && !queryString.contains("COUNT")) {
```

Comment:
@kushagraThapar for this if() statement the COUNT query that comes in fails if you add OFFSET and LIMIT, is this the correct way to handle that?

Reply:
Code removed.